### PR TITLE
fix(server): allow only one server instance for each port number

### LINKED ISSUE #268: Multiple server instances can listen to the same port
I'm using coap 0.25.0 and trying to listen to the same port using two separate Coap server instances and I though it would have thrown error when attempting to listen again on the same port using `listen(port, address, done)`. It comes out I am wrong, but is it a bug or what?

Here's a minimal sample of what I mean:
```
const coap = require("coap");

const port = 5683;
const address = "localhost";

let server = coap.createServer();
let server2 = coap.createServer();

server.once("error", (err) => {
    throw new Error(err);
});

server2.once("error", (err) => {
    throw new Error(err);
});

server.listen(port, address, (res) => {
    console.log(`error is ${res}`);
});

server2.listen(port, address, (res) => {
    console.log(`error is ${res}`);
});
```

Both the `done` callbacks correctly return and `err` is undefined. Shouldn't the second call return an error? Or I just have to assume that multiple servers can listen to connections on the same port?

### PR DESCRIPTION
**Pull request by JKRhb**
This PR would fix #268. I am not sure, however, if this change is really necessary/desirable. I am looking forward to any comments on this :) 

CC @fillobotto @relu91

### CODE REVIEW

**Review by relu91 — CHANGES_REQUESTED** [review]
Can't we expose this option somewhere? so that we can keep the current behavior but override it when it is needed?

**Comment by JKRhb** [comment]
> Can't we expose this option somewhere? so that we can keep the current behavior but override it when it is needed?

Oh, that is a very good idea! I will add it to the PR :)

**Comment by JKRhb** [comment]
@relu91 I updated the PR. Would be awesome if you could have a quick look then I'll guess this can be merged :)

**Review by relu91 — APPROVED** [review]
Looks good! Just remember to update also #278 when this lands 😄

**Comment by JKRhb** [comment]
> Looks good! Just remember to update also #278 when this lands 😄

Will do, thanks for the reminder :D

File: lib/server.js [hunk 1]
```diff
@@ -243,14 +243,18 @@ class CoAPServer extends events.EventEmitter {
 
         if (!this._options.type) this._options.type = 'udp4'
 
+        if (this._options.reuseAddr == null) {
```
Comment (Apollon77) [inline lib/server.js:246]: I would have not done the == null check that way (more like !== false ... is true :-) and only false is false ... would be my comment here :-)

**Comment by Apollon77** [comment]
see one comment, but just nitpick

File: lib/server.js (hunk 1 above)
Comment (JKRhb) [reply lib/server.js:246]: Oh, very good point! I will fix it in a follow-up PR :) Thank you for going through all of these PRs by the way! :)

Comment (Apollon77) [reply lib/server.js:246]: Was shocked by the number :-) I'm not commenting all ... just some :-)

**Pull request merged**